mtags/src/main/scala/scala/meta/internal/pc/AutoImportsProvider.scala

```diff
@@ -0,0 +1,75 @@
+package scala.meta.internal.pc
+
+import scala.meta.pc.OffsetParams
+import scala.meta.pc.AutoImportsResult
+import scala.collection.mutable
+import scala.collection.JavaConverters._
+import org.eclipse.lsp4j.TextEdit
+
+final class AutoImportsProvider(
+    val compiler: MetalsGlobal,
+    name: String,
+    params: OffsetParams
+) {
+  import compiler._
+
+  def autoImports(): List[AutoImportsResult] = {
+    val unit = addCompilationUnit(
+      code = params.text(),
+      filename = params.filename(),
+      cursor = Some(params.offset())
+    )
+    val pos = unit.position(params.offset)
+    // make sure the compilation unit is loaded
+    typedTreeAt(pos)
+    val importPosition = autoImportPosition(pos, params.text())
+    val context = doLocateImportContext(pos, importPosition)
+    val isSeen = mutable.Set.empty[String]
+    val symbols = List.newBuilder[Symbol]
+
+    def visit(sym: Symbol): Boolean = {
+      val id = sym.fullName
+      if (!isSeen(id)) {
+        isSeen += id
+        symbols += sym
+        true
+      }
+      false
+    }
+
+    val visitor = new CompilerSearchVisitor(name, context, visit)
+
+    search.search(name, buildTargetIdentifier, visitor)
+
+    def isExactMatch(sym: Symbol, name: String): Boolean =
+      sym.name.dropLocal.decoded == name
```

Comment:
this is needed since `search` above will return partially matching results, whereas we only want exact results for symbols.

@olafurpg is there a different way we could do this? I tried, but I couldn't find a way to instruct `search` to only return exact matches

Reply:
It's possible to do exact searches and the index is optimized for this use-case, I can't remember the most relevant place to look into, maybe

https://github.com/scalameta/metals/blob/24cfabf16cae3d6187a094e0711d71c125e9e79f/mtags/src/main/scala/scala/meta/internal/metals/WorkspaceSymbolQuery.scala#L35

Reply:
I saw that, but it seems `exact` still produces a list of alternatives. Anyway, it's fast enough not to be a concern (it's upper bound by completions, by definition)